internal/processor/manifests.go

```diff
@@ -224,18 +224,26 @@ func (p *Processor) validateAndStoreManifestCommon(ctx context.Context, account
 
 		// enforce account-specific validation rules on manifest, but not list manifest
 		// and only when pushing (not when validating at a later point in time,
-		// the set of RequiredLabels could have been changed by then)
-		labelsRequired := opts.IsBeingPushed && account.RequiredLabels != "" &&
+		// the validation rule could have been changed by then)
+		validationRequired := opts.IsBeingPushed && account.RuleForManifest != "" &&
 			manifest.MediaType != imageManifest.DockerV2ListMediaType && manifest.MediaType != imagespecs.MediaTypeImageIndex
-		if labelsRequired {
-			var missingLabels []string
-			for _, l := range account.SplitRequiredLabels() {
-				if _, exists := configInfo.Labels[l]; !exists {
-					missingLabels = append(missingLabels, l)
-				}
+		if validationRequired {
+			env, ast, err := keppel.BuildManifestValidationAST(account.RuleForManifest)
+			if err != nil {
+				return err
+			}
+			prg, err := env.Program(ast)
+			if err != nil {
+				return err
+			}
+			out, _, err := prg.Eval(map[string]interface{}{
+				"labels": configInfo.Labels,
+			})
+			if err != nil {
+				return err
 			}
-			if len(missingLabels) > 0 {
-				msg := "missing required labels: " + strings.Join(missingLabels, ", ")
+			if out != types.True {
+				msg := fmt.Sprintf("rule %q is not satisfied", account.RuleForManifest)
```

Comment:
For rules that are very long, this might read better as:
```suggestion
				msg := fmt.Sprintf("rule is not satisfied: %s", account.RuleForManifest)
```